upi/azure/02_storage.json

```diff
@@ -1,41 +1,153 @@
 {
-  "$schema" : "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
-  "contentVersion" : "1.0.0.0",
-  "parameters" : {
-    "baseName" : {
-      "type" : "string",
-      "minLength" : 1,
-      "metadata" : {
-        "description" : "Base name to be used in resource names (usually the cluster's Infra ID)"
+  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "architecture": {
+      "type": "string",
+      "metadata": {
+        "description": "The architecture of the Virtual Machines"
+      },
+      "defaultValue": "x64",
+      "allowedValues": [
+        "Arm64",
+        "x64"
+      ]
+    },
+    "baseName": {
+      "type": "string",
+      "minLength": 1,
+      "metadata": {
+        "description": "Base name to be used in resource names (usually the cluster's Infra ID)"
+      }
+    },
+    "storageAccount": {
+      "type": "string",
+      "metadata": {
+        "description": "The Storage Account name"
       }
     },
-    "vhdBlobURL" : {
-      "type" : "string",
-      "metadata" : {
-        "description" : "URL pointing to the blob where the VHD to be used to create master and worker machines is located"
+    "vhdBlobURL": {
+      "type": "string",
+      "metadata": {
+        "description": "URL pointing to the blob where the VHD to be used to create master and worker machines is located"
       }
     }
   },
-  "variables" : {
-    "location" : "[resourceGroup().location]",
-    "imageName" : "[concat(parameters('baseName'), '-image')]"
+  "variables": {
+    "location": "[resourceGroup().location]",
+    "galleryName": "[concat('gallery_', replace(parameters('baseName'), '-', '_'))]",
```

Comment:
How about contact 'architecture' to the galleryName? since 'arm64' and 'amd64' workers can be added to a cluster.  @r4f4

Reply:
If we do that, then we need to pass the `architecture` parameter to the VM templates (bootstrap, masters, and workers) since they reference the gallery by name.

Reply:
i would not do this since in the future when we support installing a multi-arch compute cluster through installer, we would be creating arm images under the same gallery. we don't want to be creating a separate gallery for arm instances/x86 instances. maybe the image inside the gallery could be denoted with `-arch` suffix.

Reply:
Another problem with changing the gallery name is that we also need to change https://github.com/openshift/installer/blob/master/pkg/asset/machines/azure/machines.go#L120 otherwise we break scaling with MAO.